include/grpc/impl/codegen/grpc_types.h

```diff
@@ -332,7 +333,12 @@ typedef struct grpc_op {
     struct {
       size_t count;
       grpc_metadata *metadata;
-      grpc_compression_level compression_level;
+      /** If \a is_set, \a compression_level will be used for the call.
+       * Otherwise, \a compression_level won't be considered */
+      struct {
+        bool is_set;
+        grpc_compression_level compression_level;
+      } maybe_compression_level;
```

Comment:
We reserve 8 pointers for op data (see a few lines above): we are fine here.


That said, this is a public header and must be c89. Please lose the bool.